dvc/command/completion.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import logging
+import os
+
+import shtab
+
+from dvc.command import choices
+from dvc.command.base import CmdBaseNoRepo, append_doc_link
+
+logger = logging.getLogger(__name__)
+DEFAULT_FILENAME = {"bash": "dvc", "zsh": "_dvc"}
+CHOICE_FUNCTIONS = {
+    "bash": {"DVCFile": "_dvc_compgen_DVCFiles"},
+    "zsh": {"DVCFile": "_files -g '(*.dvc|Dvcfile)'"},
+}
+PREAMBLE = {
+    "bash": """
+# $1=COMP_WORDS[1]
+_dvc_compgen_DVCFiles() {
+  compgen -f -X '!*?.dvc' -- $1  # DVC-files
+  compgen -d -S '/' -- $1  # recurse into subdirs
+}
+""",
+    "zsh": "",
```

Comment:
Missing the new `dvc.yaml` files.

Reply:
Hmm so we need:

- *.dvc
- Dvcfile
- dvc.yaml
- dvc.yml

and potentially case-insensitivity?

Reply:
@casperdcl No, just

* *.dvc
* Dvcfile
* dvc.yaml

since those are the only *officially* supported ones. We'll add others later if we need to.

Reply:
@casperdcl Also, command like `repro` should support getting particular stage name as an argument. We don't have a command to list stages right now in a form that would be useful in the completion, but just saying that we should be aware of that.

Reply:
yes I was thinking about this. The way it's implemented now it'll be trivial to add (assuming something like `dvc stage list` will be provided in future)

Reply:
updated